packages/bot-web-ui/src/components/dashboard/dbot-tours/mobile-tours/__tests__/bot-builder-tour.spec.tsx

```diff
@@ -0,0 +1,122 @@
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import BotBuilderTour from '../bot-builder-tour';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+jest.mock('@deriv/deriv-charts', () => ({
+    setSmartChartsPublicPath: jest.fn(),
+}));
+
+describe('<BotBuilderTour />', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('renders botbuilder tour', () => {
+        const { container } = render(<BotBuilderTour />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+    });
+
+    it('shows the tour content when tour starts', () => {
+        render(<BotBuilderTour />, {
+            wrapper,
+        });
+        const tourElement = screen.getByTestId('botbuilder-tour-mobile');
+        expect(tourElement).toHaveClass('dbot-slider__bot-builder-tour');
+    });
+
+    it('Check if next button is clicked', async () => {
+        render(<BotBuilderTour />, {
+            wrapper,
+        });
+        const nextButton = screen.getByTestId('next-bot-builder-tour');
+        if (!nextButton) {
+            console.error('Next button not found. Make sure it exists in your component.'); // eslint-disable-line no-console
+        }
+        const onClickMock = jest.fn();
+        nextButton.onclick = onClickMock;
+        act(() => {
+            userEvent.click(nextButton);
+        });
+        await waitFor(() => {
+            expect(onClickMock).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    it('should check Previous button is clicked', async () => {
+        render(<BotBuilderTour />, {
+            wrapper,
+        });
+        const nextButton = screen.getByTestId('next-bot-builder-tour');
+        const nextButtonOnClickMock = jest.fn();
+        nextButton.onclick = nextButtonOnClickMock;
+        act(() => {
+            userEvent.click(nextButton);
+        });
+        const prevButton = screen.getByTestId('prev-bot-builder-tour');
+        const prevButtonOnClickMock = jest.fn();
+        prevButton.onclick = prevButtonOnClickMock;
+        act(() => {
+            userEvent.click(prevButton);
+        });
+        await waitFor(() => {
+            expect(nextButtonOnClickMock).toHaveBeenCalledTimes(1);
+            expect(prevButtonOnClickMock).toHaveBeenCalledTimes(1);
```

Comment:
Same here. Can we check the effect of the button click instead of checking whether the button is clicked or not?